plugins/idea/src/main/java/com/oracle/plugin/jtreg/configuration/producers/JTRegConfigurationProducer.java

```diff
@@ -65,10 +67,95 @@ public boolean isConfigurationFromContext(JTRegConfiguration unitConfiguration,
-                runClass.equals(testClass.getVirtualFile().getPath());
+
+        PsiElement element = location.getPsiElement();
+        String query = getQuery(element);
+        PsiFile contextFile = element.getContainingFile();
+        final String configFile = unitConfiguration.getRunClass();
+        return configFile != null && contextFile != null && contextFile.getVirtualFile() != null &&
+                configFile.equals(contextFile.getVirtualFile().getPath())
+                && query.equals(unitConfiguration.getQuery());
+    }
+
+    protected static String getQuery(PsiElement element) {
+        boolean isJUnit = JTRegUtils.isJUnitTestData(element.getContainingFile());
+        boolean isTestNG = JTRegUtils.isTestNGTestData(element.getContainingFile());
+        if (element instanceof PsiIdentifier
+                && element.getParent() instanceof PsiMethod method) {
+            if (isJUnit) {
+                return "junit-select:method:" + getJUnitMethodQuery(method);
+            } else if (isTestNG) {
+                // just the method name for TestNG
+                return method.getName();
+            }
+        } else if (isJUnit && element instanceof PsiIdentifier
+                && element.getParent() instanceof PsiClass cls) {
+            return "junit-select:class:" + binaryNameFor(cls);
+        }
+        return "";
+    }
+
+    private static String getJUnitMethodQuery(PsiMethod method) {
+        StringJoiner paramTypeNames = new StringJoiner(",");
+        PsiParameterList params = method.getParameterList();
+        for (int i = 0; i < params.getParametersCount(); i++) {
+            PsiParameter param = params.getParameter(i);
+            PsiType type = param.getType();
+            paramTypeNames.add(binaryNameFor(type));
+        }
+        String className = binaryNameFor(((PsiClass) method.getParent()));
+        return className + "#" + method.getName() + '(' + paramTypeNames + ')';
+    }
+
+    private static String binaryNameFor(PsiType type) {
+        if (type instanceof PsiPrimitiveType) {
+            return type.getCanonicalText();
+        } else if (type instanceof PsiClassType clsType) {
+            return binaryNameFor(clsType.resolve());
+        } else if (type instanceof PsiArrayType arrayType) {
+            return binaryNameFor(arrayType);
+        } else {
+            return type.getCanonicalText();
+        }
+    }
+
+    private static String binaryNameFor(PsiClass cls) {
+        // handle nested classes. Convert '.' to '$'
+        String nestedName = cls.getName();
+        PsiClass current = cls;
+        while ((current = current.getContainingClass()) != null) {
+            nestedName = current.getName() + "." + nestedName;
+        }
+        String qualName = cls.getQualifiedName();
+        String packageName = qualName.substring(0, qualName.length() - nestedName.length());
+        return packageName + nestedName.replace('.', '$');
```

Comment:
Boy this is pretty ugly (not your fault - can't find a way to get package from class)